projects/plugins/boost/app/modules/critical-css/class-critical-css.php

```diff
@@ -652,31 +654,76 @@ public function get_critical_css() {
 	 * @see style_loader_tag
 	 */
 	public function asynchronize_stylesheets( $html, $handle, $href, $media ) {
+		// If is AMP, do not alter the stylesheet loading.
 		if ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ) {
 			return $html;
 		}
 
+		// If there is no critical CSS, do not alter the stylesheet loading.
 		if ( false === $this->get_critical_css() ) {
 			return $html;
 		}
 
-		if ( ! apply_filters( 'jetpack_boost_async_style', true, $handle ) ) {
+		$available_methods = array(
+			'instant'  => 'media="' . $media . '"',
+			'async'    => 'media="not all" onload="this.media=\'all\'"',
+			'deferred' => 'media="not all"',
+		);
+
+		/**
+		 * Loading method for stylesheets.
+		 *
+		 * Filter the loading method for each stylesheet for the screen with following values:
+		 *     instant  - Stylesheet loading behaviour is not altered. Stylesheet loading is instant and the process blocks the page rendering.
+		 *     async    - Stylesheets are loaded asynchronously. Styles are applied once the stylesheet is loaded completely without render blocking.
+		 *     deferred - Loading of stylesheets are deferred till the window load event. Styles from all the stylesheets are applied at once after the page load.
+		 *
+		 * @see onload_flip_stylesheets for how stylesheets loading is deferred.
+		 *
+		 * @param string $handle The style's registered handle.
+		 * @param string $media  The stylesheet's media attribute.
+		 *
+		 * @todo  Retrieve settings from database, either via auto-configuration or UI option.
+		 */
+		$method = apply_filters( 'jetpack_boost_async_style', 'async', $handle, $media );
```

Comment:
And to change the filter name from `jetpack_boost_async_style` to `jetpack_boost_style_loading_method` since this is a filter to change the stylesheet loading behaviour